mdpow/tests/test_automated_dihedral_analysis.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import absolute_import
+
+import os
+import sys
+import yaml
+import pybol
+import pytest
+import pathlib
+
+import scipy
+import numpy as np
+import pandas as pd
+
+import rdkit
+from rdkit import Chem
+
+from numpy.testing import assert_almost_equal
+from scipy.stats import circvar, circmean
+
+from . import RESOURCES
+
+import py.path
+
+from ..workflows import dihedrals
+
+from pkg_resources import resource_filename
+
+RESOURCES = pathlib.PurePath(resource_filename(__name__, 'testing_resources'))
+MANIFEST = RESOURCES / "manifest.yml"
+
+@pytest.fixture(scope="function")
+def molname_workflows_directory(tmp_path, molname='SM25'):
+    m = pybol.Manifest(str(RESOURCES / 'manifest.yml'))
+    m.assemble('workflows', tmp_path)
+    return tmp_path / molname
+
+class TestAutomatedDihedralAnalysis(object):
+
+    @pytest.fixture(scope="function")
+    def SM25_tmp_dir(self, molname_workflows_directory):
+        dirname = molname_workflows_directory
+        return dirname
+
+    @pytest.fixture(scope="function")
+    def gen_data(self, SM25_tmp_dir):
+        atom_group_indices = dihedrals.dihedral_indices(dirname=SM25_tmp_dir, resname=self.resname)
+        # test user input of SMARTS string
+        atom_group_indices_alt = dihedrals.dihedral_indices(dirname=SM25_tmp_dir, resname=self.resname,
+                                                            SMARTS='[!$(*#*)&!D1]-!@[!$(*#*)&!D1]')
```

Comment:
All functions that depend on `dihedral_indices` function are by default done with our SMARTS string that returns four-tuples of dihedral atom groups. I will add comments to address these differences in the data, fixtures, and tests.

This alternative SMARTS string is the original provided by Dr. Iorga that I built upon to extend from two-tuples to four-tuples; this one returns the two-tuples. I believe it is a good example for testing the MDA/RDKit conversion functionality because we are expecting that the 1-2 indices of the four-tuples should be the same as the two-tuples. Also, I can add an `assert` line or a separate test to verify this and really make sure that MDA-RDKit-SMARTS is doing what we want it to all the time.